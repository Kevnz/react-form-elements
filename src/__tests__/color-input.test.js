import React from 'react'
import { render, cleanup, fireEvent } from 'react-testing-library'

import ColorInput from '../color-input'

describe('ColorInput component', () => {
  afterEach(() => {
    cleanup()
  })
  describe('The Rendered HTML', () => {
    it('should have a label', () => {
      const { getByText, container } = render(
        <ColorInput label="My ColorInput Label" />
      )

      expect(container.firstChild).toMatchSnapshot()
      const label = getByText('My ColorInput Label')
      expect(label.textContent).toBe('My ColorInput Label')
    })
  })
  describe('handling input', () => {
    it('should set the value to `#000000` when passed non color input', () => {
      const { getByLabelText, container } = render(
        <ColorInput label="My Color Value" initialValue="#0668fa" />
      )

      expect(container.firstChild).toMatchSnapshot()
      const input = getByLabelText('My Color Value')
      expect(input.value).toBe('#0668fa')
      fireEvent.change(input, { target: { value: 'blue' } })
      expect(input.value).toBe('#000000')
    })
  })
})
