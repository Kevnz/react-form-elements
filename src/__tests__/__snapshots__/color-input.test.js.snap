// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorInput component The Rendered HTML should have a label 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-1-1"
  >
    My ColorInput Label
  </label>
  <input
    id="id-1-1"
    name="TextBox"
    type="color"
    value=""
  />
</div>
`;

exports[`ColorInput component handling input should set the value to \`#000000\` when passed non color input 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-2-1"
  >
    My Color Value
  </label>
  <input
    id="id-2-1"
    name="TextBox"
    type="color"
    value="#0668fa"
  />
</div>
`;
